docs/DeveloperGuide.md

```diff
@@ -776,6 +739,141 @@ have already gotten their house.
 
       Use case ends.
 
+**Use case: UC13 - Find matching sellers for a buyer**
+
+**MSS:**
+1. User requests to <u>filter out buyers (UC12)</u>.
+2. User requests to find all the matching sellers for a buyer based on their requirements.
+3. EstateEase displays the list of sellers who have properties that match the buyer's requirements.
+
+Use case ends.
+
+**Extensions**
+
+* 1a. The list is empty.
+  
+  Use case ends.
+
+* 2a. The given index for the buyer is invalid input type.
+    * 2a1. EstateEase shows an error message indicating the invalid input type.
+      
+      Use case resumes at step 1.
+
+* 2b. The given index for the buyer is out of range.
+    * 2b1. EstateEase shows an error message indicating the out of range for the index.
+      
+      Use case resumes at step 1.
+
+* 3a. There are no sellers in the contact list.
+    * 3a1. EstateEase shows a message indicating there is no sellers in the contact list.
+      
+      Use case ends.
+
+* 3b. There are no matching properties based on the buyer's requirements.
+    * 3b1. EstateEase shows a message indicating there is no matching results.
+      
+      Use case ends.
+
+
+**Use case: UC14 - Link Buyer to Seller**
+
+**MSS:**
+
+1. User initiates the process of linking a buyer to sellers for a specific property.
+2. EstateEase validates the provided property information, buyer ID, and seller ID.
+3. EstateEase proceeds to link the buyer to the specified seller(s) for the given property.
+4. Use case ends.
+
+**Extensions**
+
+* 1a. User provides invalid input for linking.
+    * 1a1. EstateEase displays an error message indicating the issue with the input.
+      Use case ends.
+
+
+**Use case: UC15 - View home sellers by priority**
+
+**Preconditions:**
+- The user initialises a view command with home-seller as a filter
+
+**MSS:**
+1.  EstateEase process the view command with home-seller as filter.
+2.  EstateEase shows a list of home-sellers, arranged based on their priority. <br>
+    Use case ends.
+
+**Extensions**
+* 1a. The contact list does not have any home-seller. <br>
+    * 1a1. EstateEase shows an error message stating that the contact list does not have home-seller. <br>
+      Use case ends.
+
+      
+
+**Use case: UC16 - Differentiate home-seller status**
+
+**MSS:**
+
+1. User filters for home-sellers
+2. EstateEase displays home-sellers. Free home-sellers are highlighted in green.
+
+    Use case ends.
+
+
+**Extensions**
+
+* 2a. Pending home-sellers are displayed in red.
```

Comment:
I am not too sure whether needs to be very UI-specified, because I remember use case should not include the UI specification (so maybe you can check on this?)